x-pack/plugins/maps/server/data_indexing/index_data.ts

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { i18n } from '@kbn/i18n';
+import { ElasticsearchClient } from 'kibana/server';
+import { WriteSettings } from '../../common';
+
+export async function writeDataToIndex(
+  index: string,
+  data: object,
+  asCurrentUser: ElasticsearchClient
+) {
+  try {
```

Comment:
similar to our existing routes, we'll need to do some explicit user-permission checking. I understand the how-when is still in flux, but are we tracking this as a to-do?

Reply:
It's in flight now: https://github.com/elastic/kibana/pull/95190